troposphere/ec2.py

```diff
@@ -761,3 +762,58 @@ class VPCCidrBlock(AWSObject):
         'CidrBlock': (basestring, False),
         'VpcId': (basestring, True),
     }
+
+
+class TagSpecifications(AWSProperty):
+    props = {
+        'ResourceType': (basestring, False),
+        'Tags': ((Tags, list), False)
+    }
+
+
+class SpotOptions(AWSProperty):
+    props = {
+        'InstanceInterruptionBehavior': (basestring, False),
+        'MaxPrice': (basestring, False),
+        'SpotInstanceType': (basestring, False)
+    }
+
+
+class InstanceMarketOptions(AWSProperty):
+    props = {
+        'MarketType': (basestring, False),
+        'SpotOptions': (SpotOptions, False)
+    }
+
+
+class LaunchTemplateData(AWSProperty):
+    props = {
+        'BlockDeviceMappings': ([BlockDeviceMapping], False),
+        'CreditSpecification': (CreditSpecification, False),
+        'DisableApiTermination': (boolean, False),
+        'EbsOptimized': (boolean, False),
+        'ElasticGpuSpecifications': ([ElasticGpuSpecification], False),
+        'IamInstanceProfile': (IamInstanceProfile, False),
+        'ImageId': (basestring, True),
```

Comment:
Strangely, this doesn't actually appear to be required per the docs.

Reply:
I verified through CFN execution that both InstanceType and ImageId are required

Reply:
I have direct access to the CFN team, I am going to get a call setup with them next week and get clarity on all these namespacing issues.

Reply:
Awesome, thanks @boostchicken